### PR TITLE
th/hexstr2bin-cleanup

### PR DESCRIPTION
**Pull request by thom311**
Various cleanup of our hexstr <-> bin conversions.

### CODE REVIEW

**Comment by thom311** [comment]
The

```
+    //FIXME: verify that priv->mka_cak is a valid hex-string.
```
in libnm-core/nm-setting-macsec.c is unclear to me how to fix. @bengal ?

**Comment by thom311** [comment]
> The
> 
> ```
> +    //FIXME: verify that priv->mka_cak is a valid hex-string.
> ```
> 
> in libnm-core/nm-setting-macsec.c is unclear to me how to fix. @bengal ?

Done now with "libnm/macsec: tighten up verification and normalize mka_cak/mka_ckn properties"

File: libnm-core/nm-setting-macsec.c [hunk 1]
```diff
@@ -352,6 +356,13 @@ verify (NMSetting *setting, NMConnection *connection, GError **error)
 			g_prefix_error (error, "%s: ", NM_SETTING_MACSEC_SETTING_NAME);
 			return FALSE;
 		}
+	} else {
+		g_set_error_literal (error,
+		                     NM_CONNECTION_ERROR,
+		                     NM_CONNECTION_ERROR_MISSING_SETTING,
```
Comment (bengal) [inline libnm-core/nm-setting-macsec.c]: NM_CONNECTION_ERROR_INVALID_PROPERTY ?

File: libnm-core/nm-setting-macsec.c [hunk 2]
```diff
@@ -364,6 +375,17 @@ verify (NMSetting *setting, NMConnection *connection, GError **error)
 		return FALSE;
 	}
 
+	if (   priv->mode != NM_SETTING_MACSEC_MODE_PSK
+	    && (priv->mka_cak || priv->mka_ckn)) {
+		g_set_error (error,
+		             NM_CONNECTION_ERROR,
+		             NM_CONNECTION_ERROR_MISSING_SETTING,
```
Comment (bengal) [inline libnm-core/nm-setting-macsec.c]: NM_CONNECTION_ERROR_INVALID_PROPERTY ?

File: libnm-core/nm-setting-macsec.c [hunk 3]
```diff
@@ -364,6 +375,17 @@ verify (NMSetting *setting, NMConnection *connection, GError **error)
...
+		             _("%s only valid for psk mode"),
+		             priv->mka_cak ? NM_SETTING_MACSEC_MKA_CAK : NM_SETTING_MACSEC_MKA_CKN);
+		g_prefix_error (error, "%s: ", NM_SETTING_MACSEC_SETTING_NAME);
```
Comment (bengal) [inline libnm-core/nm-setting-macsec.c]: Can we keep the usual syntax like "macsec.cak: only valid for psk mode", i.e.:

```
      g_set_error_literal (error,
                                   NM_CONNECTION_ERROR,
                                   NM_CONNECTION_ERROR_INVALID_PROPERTY,
                                   _("only valid for psk mode"));
       g_prefix_error (error, "%s.%s:", NM_SETTING_MACSEC_SETTING_NAME,
                              priv->mka_cak ? NM_SETTING_MACSEC_MKA_CAK : NM_SETTING_MACSEC_MKA_CKN);  
```

File: libnm-core/nm-connection.c [hunk 4]
```diff
@@ -1124,6 +1124,29 @@ _normalize_wireless_mac_address_randomization (NMConnection *self, GHashTable *p
...
+static gboolean
+_normalize_macsec (NMConnection *self, GHashTable *parameters)
+{
+	NMSettingMacsec *s_macsec = nm_connection_get_setting_macsec (self);
+	gboolean changed = FALSE;
+
+	if (!s_macsec)
+		return FALSE;
+
+	if (nm_setting_macsec_get_mode (s_macsec) == NM_SETTING_MACSEC_MODE_PSK) {
```
Comment (bengal) [inline libnm-core/nm-connection.c]: mode != PSK

**Comment by thom311** [comment]
merged

**Pull request merged**